test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,173 @@
+// Special Methods
+
+/* This primer covers special methods for classes and records */
+
+// Classes and records can have a handful of methods that are specially named
+// and treated differently than other methods.  These methods enable
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Throughout this primer, the methods are defined as secondary methods
+// using the form: ``proc TypeName.methodName() ...``
```

Comment:
I know it might seem irrelevant but I don't think "primary" and "secondary" methods are well known terms and I don't think the examples are detailed enough. Could you say something like:

Note that there are two ways to declare a method. Methods declared within a class or record are called primary methods:
``` chapel
record ExampleRecord1 {
  var exampleField:int;
  proc someMethod() { }
}
```

Methods declared outside of a class are called secondary methods:
``` chapel
record ExampleRecord2 {
  var exampleField:int;
}
proc ExampleRecord2.someMethod() { }
```

This primer will use the secondary method for but all of the special methods can also be written as primary methods.